### PR TITLE
[INT-189] Automatically bypass login page if session exists

### PR DESCRIPTION
**Pull request by archessmn**

### CODE REVIEW

File: app/login/page.tsx [hunk 1]
```diff
@@ -6,10 +6,13 @@ import SlackLoginButton from "@/components/slack/SlackLoginButton";
...
 import { Center, Stack } from "@mantine/core";
 import { env } from "@/lib/env";
+import { mustNotGetSession } from "@/lib/auth/server";
 
-export default function GoogleSignInPage(props: {
-  searchParams: { error?: string };
+export default async function GoogleSignInPage(props: {
+  searchParams: { error?: string; redirect?: string };
 }) {
+  await mustNotGetSession(undefined, props.searchParams.redirect);
```
Comment (markspolakovs) [inline app/login/page.tsx:14]: nitpick: when I first see `must` I imagine "will fail if it does" - perhaps something like `ensureNoActiveSession`?

File: lib/auth/server.ts [hunk 2]
```diff
@@ -134,6 +134,20 @@ export async function mustGetCurrentUser(req?: NextRequest): Promise<UserType> {
   return userOrError;
 }
 
+// Redirects if there is a user logged in already
+export async function mustNotGetSession(
+  req?: NextRequest,
```
Comment (markspolakovs) [inline lib/auth/server.ts:139]: is this parameter used?

File: app/login/page.tsx (hunk 1 above)
Comment (archessmn) [reply app/login/page.tsx]: Updated

File: lib/auth/server.ts (hunk 2 above)
Comment (archessmn) [reply lib/auth/server.ts]: Ah forgot to remove that when I copied bits

**Review by markspolakovs — APPROVED** [review]

**Pull request merged**